### ISSUE TITLE
Unable to upload new version of compliance profile

### ISSUE THREAD

**Issue by danielcbright** [inspec/inspec#3137]
### 🎛 Description

🐛 bug report

When uploading a newer version of a custom compliance profile, with a new version defined in `inspec.yml` I expect it to upload the newer version of the profile to the A2 Compliance store and for both versions to be available. What's actually happening is I can't upload the profile. I get the following error:

```
inspec compliance upload my-linux-baseline --no-overwrite

Profile is already vendored. Use --overwrite.
I, [2018-06-13T10:53:20.141099 #69172]  INFO -- : Checking profile in my-linux-baseline
I, [2018-06-13T10:53:20.143124 #69172]  INFO -- : Metadata OK.
I, [2018-06-13T10:53:20.303109 #69172]  INFO -- : Found 7 controls.
I, [2018-06-13T10:53:20.303189 #69172]  INFO -- : Control definitions OK.
Profile is valid
Profile exists on the server, use --overwrite
Found 1 error(s)
```

### 🌍 InSpec and Platform Version

Inspec Version: `2.1.72`
Mac OS `10.13.4`

### 🤔 Replication Case

1. Create a new custom profile with a version and `inspec compliance upload` that profile
2. Update the profile and bump the version number in `inspec.yml` and then attempt to upload the profile again, you can only do so if you overwrite.

**Comment by clintoncwolfe** [comment]
Verified, the clobber check does not ask about version - see here https://github.com/inspec/inspec/blob/33fc15582ad8134cb6d948bb59b622a34dc62454/lib/bundles/inspec-compliance/cli.rb#L178

**Comment by danielcbright** [comment]
Interesting note, when I do specify `--overwrite`, it uploads the profile with the new version, and actually doesn't overwrite the existing one, so I end up with the behavior I wanted in the first place, just not with the right option.

**Comment by frezbo** [comment]
`--overwrite`  is for lockfile updates AFAIK

**Comment by clintoncwolfe** [comment]
Fixed by #3252
